### PR TITLE
Disable number grouping (thousands comma separators) by default in Vue node number widgets

### PR DESCRIPTION
**Pull request by christian-byrne**
## Summary

Makes the [useGrouping](https://primevue.org/inputnumber/#api.inputnumber.props.useGrouping) prop for number widgets disabled by default, aligning with the old UI (also requested via design). Node authors can still enable if they want by setting prop explicitly.

## Changes

- **What**: Modified [WidgetInputNumberInput](https://primevue.org/inputnumber/) to disable `useGrouping` by default, requiring explicit opt-in via widget options
- **Testing**: Added component tests covering value binding, component rendering, step calculations, and grouping behavior

## Review Focus

UX impact on existing nodes that may have relied on default grouping behavior and test coverage for edge cases with precision calculations.

## Screenshots (if applicable)

*Before*:

Screenshot from 2025-09-25 11-34-34

*After*:

Screenshot from 2025-09-25 11-35-27

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-5776-Disable-number-grouping-thousands-comma-separators-by-default-in-Vue-node-number-widget-2796d73d365081369ca6c155335d0d57) by [Unito](https://www.unito.io)

### CODE REVIEW

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    props: {
+      widget,
+      modelValue,
+      readonly
+    }
+  })
+}
+
+function getNumberInput(wrapper: ReturnType<typeof mount>) {
+  const input = wrapper.find('input[inputmode="numeric"]')
```
Comment (DrJKL) [inline src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts]: If you use [get](https://test-utils.vuejs.org/api/#get) instead of find, it'll throw for you.

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    const inputNumber = wrapper.findComponent(InputNumber)
+    expect(inputNumber.props('useGrouping')).toBe(false)
+  })
+
+  it('enables grouping when explicitly set to true in widget options', () => {
+    const widget = createMockWidget(1000, 'int', { useGrouping: true })
+    const wrapper = mountComponent(widget, 1000)
+
+    const inputNumber = wrapper.findComponent(InputNumber)
+    expect(inputNumber.props('useGrouping')).toBe(true)
```
Comment (DrJKL) [inline src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts]: A lot of these feel like they're just testing Vue's binding system.
What about checking the rendered text for commas instead?

**Review by DrJKL — DISMISSED** [review]

**Review by AustinMroz — COMMENTED** [review]
LGTM.

Question: Since useGrouping is a new option, should it be added to `litegraph/src/types/widgets.ts`? Do we intend to keep these litegraph interfaces up to date/move them out of the litegraph directory?

**Comment by christian-byrne** [comment]
Test failures are expected due to the seed display in default workflow losing the thousands comma separators: https://59ad2721.comfyui-playwright-chromium.pages.dev/#?q=s:failed

**Comment by christian-byrne** [comment]
> Question: Since useGrouping is a new option, should it be added to litegraph/src/types/widgets.ts? Do we intend to keep these litegraph interfaces up to date/move them out of the litegraph directory?

I don't think we need to update the litegraph interfaces, as litegraph won't be doing anything with the new widget options. The types will be defined in python -> node schemas -> Vue component props.

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts (hunk 1 above)
Comment (christian-byrne) [reply src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts]: Switched to wrapper.get() for better error messages in 2e6047b

Comment (christian-byrne) [reply src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts]: Done https://github.com/Comfy-Org/ComfyUI_frontend/pull/5776/commits/2e6047bbf2452e55c16a6fcae01f39700586583c

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts (hunk 2 above)
Comment (christian-byrne) [reply src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts]: Done https://github.com/Comfy-Org/ComfyUI_frontend/pull/5776/commits/df3c5acf27bcd462d34c365ae4908203f63c3027

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  })
+
+  it('displays numbers with commas when grouping enabled', () => {
+    const widget = createMockWidget(1000, 'int', { useGrouping: true })
+    const wrapper = mountComponent(widget, 1000)
+
+    const input = getNumberInput(wrapper)
+    expect(input.value).toBe('1,000')
+    expect(input.value).toContain(',')
+  })
```
Comment (DrJKL) [inline src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts:189]: idle thought: Very much not something I think is important to test right now, but does the PrimeVue component handle the internationalization of the separator?

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+function getNumberInput(wrapper: ReturnType<typeof mount>) {
+  const input = wrapper.get('input[inputmode="numeric"]')
+  if (!(input.element instanceof HTMLInputElement)) {
+    throw new Error(
+      'Number input element not found or is not an HTMLInputElement'
+    )
+  }
```
Comment (DrJKL) [inline src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts]:
```suggestion
  const input = wrapper.get<HTMLInputElement>('input[inputmode="numeric"]')
```

**Review by DrJKL — DISMISSED** [review]

File: src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts (hunk 3 above)
Comment (christian-byrne) [reply src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.test.ts:189]: There's a `locale` prop https://primevue.org/inputnumber/#api.inputnumber.props.locale. I'm assuming they have intelligent default.

**Comment by christian-byrne** [comment]
As mentioned [here](https://github.com/Comfy-Org/ComfyUI_frontend/pull/5776#issuecomment-3336090481), test failure is expected now that comma is gone, so re-generating baseline.
